### PR TITLE
refactor: extract generic Registry base class for ActionRegistry

### PR DESCRIPTION
**Pull request by mthalman**
- Disposal callback now runs at most once via a captured disposed flag
- Prevents stale Disposable from removing a re-registered same-instance entry
- Add test for dispose-reregister-same-instance-dispose edge case

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: packages/core/src/services/registry.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  constructor(kind: string) { this.kind = kind; }
+
+  register(item: T): vscode.Disposable {
+    if (this.items.has(item.id)) {
+      throw new Error(`${this.kind} already registered: ${item.id}`);
+    }
+    this.items.set(item.id, item);
+    logger.info(`Registered ${this.kind.toLowerCase()}: ${item.id} (${item.label})`);
+    return new vscode.Disposable(() => { this.items.delete(item.id); });
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/services/registry.ts]: Good catch! Fixed in e73e83e - the Disposable now checks identity before deleting, making disposal safe across re-registrations.

File: packages/core/src/test/registry.test.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+  it('allows re-registration after disposal', () => {
+    const disposable = registry.register(createItem('reuse'));
+    disposable.dispose();
+    const item2 = createItem('reuse', 'Reused');
+    registry.register(item2);
+    expect(registry.get('reuse')).toBe(item2);
+  });
+
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/registry.test.ts:74]: Added this test in e73e83e. It validates that disposing the old Disposable after re-registration doesn't affect the new item.

File: packages/core/src/services/registry.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    if (this.items.has(item.id)) {
+      throw new Error(`${this.kind} already registered: ${item.id}`);
+    }
+    this.items.set(item.id, item);
+    logger.info(`Registered ${this.kind.toLowerCase()}: ${item.id} (${item.label})`);
+    return new vscode.Disposable(() => {
+      if (this.items.get(item.id) === item) {
+        this.items.delete(item.id);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/services/registry.ts:22]: Fixed in 8c7c118 - added a captured \disposed\ flag so the callback runs at most once, making it safe even when the same instance is re-registered.
